combat-trainer.lic

```diff
@@ -75,55 +63,29 @@ class SetupProcess
       check_stance(game_state)
     end
 
-    game_state.new_weapon_skill(game_state.weapon_skill)
+    game_state.update_weapon_skill(game_state.weapon_skill)
     false
   end
 
   private
 
-  def dance(game_state)
-    game_state.new_weapon_skill(@dance_skill)
-    game_state.weapon_name = @weapon_training[@dance_skill]
-  end
-
-  def retreat_weapons(weapon_training)
-    weapon_training.each_with_object({}) { |info, obj| obj[info.first] = info.last if $ranged_skills.include?(info.first) }
-  end
-
-  def skill_done?(game_state)
-    game_state.action_count >= @target_action_count || DRSkill.getxp(game_state.weapon_skill).to_i >= game_state.target_weapon_skill
-  end
-
-  def ending_ranged?(game_state)
-    game_state.was_retreating && !game_state.now_retreating
-  end
-
   def determine_next_to_train(game_state, weapon_training)
-    return unless skill_done?(game_state) || !weapon_training[game_state.weapon_skill] || ending_ranged?(game_state)
+    return unless game_state.skill_done?(DRSkill.getxp(game_state.weapon_skill).to_i) || !weapon_training[game_state.weapon_skill] || game_state.ending_ranged?
```

Comment:
It knows that there is a number it needs to compare, but not how that number is generated. Maybe that's a meaningless distinction? @kamarcum thoughts?

Will having GameState use DRSkill make that method any harder to test?